src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -648,29 +645,29 @@ def _format_entry_points(
             for category, entry_points in all_entry_points.items()
         }
 
-    # Generate the setup script.
-    setup_py_content = SETUP_BOILERPLATE.format(
-        target_address_spec=target.address.spec,
-        setup_kwargs_str=distutils_repr(setup_kwargs),
-    ).encode()
-    files_to_create = [
-        FileContent("setup.py", setup_py_content),
-        FileContent("MANIFEST.in", b"include *.py"),
-    ]
-    extra_files_digest = await Get(Digest, CreateDigest(files_to_create))
-    chroot_digest = await Get(Digest, MergeDigests((sources.digest, extra_files_digest)))
-    return SetupPyChroot(
-        chroot_digest, "setup.py", FinalizedSetupKwargs(setup_kwargs, address=target.address)
-    )
+    return FinalizedSetupKwargs(setup_kwargs, address=target.address)
 
 
 @rule
-async def get_sources(request: SetupPySourcesRequest) -> SetupPySources:
+async def get_sources(request: SetupPyChrootRequest) -> SetupPySources:
+    owned_deps, transitive_targets = await MultiGet(
+        Get(OwnedDependencies, DependencyOwner(request.exported_target)),
+        Get(
+            TransitiveTargets,
+            TransitiveTargetsRequest([request.exported_target.target.address]),
+        ),
+    )
+    # files() targets aren't owned by a single exported target - they aren't code, so
```

Comment:
This seems good for your example but bad for any example where the files() end up in a package prefix of the dist (effectively netting you a resource that's dup in different dists). I'm not sure that's easy to calculate though and I'm equally unsure how much effort it's worth to anticipate.

Reply:
Yeah, this might be a small footgun that we leave in for now...